### PR TITLE
Filling in empty doc comments.

### PR DESCRIPTION
**Pull request by mitchdenny**
This is a quick pass over all of our docs trying to fill in some blanks. There are more for sure but this is a start.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2968)

### CODE REVIEW

File: src/Aspire.Hosting.AWS/CloudFormation/CloudFormationStackExecutor.cs [hunk 1]
```diff
@@ -386,8 +386,8 @@ private async Task<bool> WaitForChangeSetBeingAvailableAsync(string changeSetId,
     /// <summary>
     /// Wait for the CloudFormation stack to get to a stable state after creating or updating the stack.
     /// </summary>
-    /// <param name="minTimeStampForEvents"></param>
-    /// <param name="cancellationToken"></param>
+    /// <param name="minTimeStampForEvents">Minimum timestamp for events.</param>
+    /// <param name="cancellationToken">A cancellation token</param>
```
Comment (JamesNK) [inline src/Aspire.Hosting.AWS/CloudFormation/CloudFormationStackExecutor.cs:390]:
```suggestion
    /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
```

File: src/Aspire.Hosting.AWS/SDKResourceExtensions.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public static IAWSSDKConfig WithRegion(this IAWSSDKConfig config, RegionEndpoint
     /// <summary>
     /// Add a reference to an AWS SDK configuration a project.
     /// </summary>
-    /// <param name="builder"></param>
+    /// <param name="builder">A <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
```
Comment (JamesNK) [inline src/Aspire.Hosting.AWS/SDKResourceExtensions.cs:53]:
```suggestion
    /// <param name="builder">An <see cref="IResourceBuilder{T}"/> for <see cref="ProjectResource"/></param>
```

File: src/Aspire.Hosting.Azure/AzureBicepResource.cs [hunk 3]
```diff
@@ -127,12 +127,13 @@ public virtual string GetBicepTemplateString()
         return File.ReadAllText(TemplateFile);
     }
 
-    // TODO: Make the name bicep safe
+    /// TODO: Remove this method once AppInsights CDK is removed.
     /// <summary>
-    /// TODO: Doc Comments
+    /// Create a Bicep identifier safe version of the resource name.
     /// </summary>
     /// <returns></returns>
```
Comment (JamesNK) [inline src/Aspire.Hosting.Azure/AzureBicepResource.cs:134]: Empty returns

File: src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs [hunk 4]
```diff
@@ -168,11 +168,11 @@ public static IResourceBuilder<AzureStorageEmulatorResource> UseTablePort(this I
...
     /// <summary>
-    /// TODO: Doc Comments
+    /// Creates a builder for the <see cref="AzureBlobStorageResource"/> which can be referenced to get the Azure Storage blob endpoint for the storage account.
     /// </summary>
-    /// <param name="builder"></param>
-    /// <param name="name"></param>
-    /// <returns></returns>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>/</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns>A <see cref="IResourceBuilder{T}"/> for the <see cref="AzureBlobStorageResource"/>.</returns>
```
Comment (JamesNK) [inline src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs:175]:
```suggestion
    /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureBlobStorageResource"/>.</returns>
```

File: src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs [hunk 5]
```diff
@@ -182,11 +182,11 @@ public static IResourceBuilder<AzureBlobStorageResource> AddBlobs(this IResource
...
     /// <summary>
-    /// TODO: Doc Comments
+    /// Creates a builder for the <see cref="AzureTableStorageResource"/> which can be referenced to get the Azure Storage tables endpoint for the storage account.
     /// </summary>
-    /// <param name="builder"></param>
-    /// <param name="name"></param>
-    /// <returns></returns>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>/</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns>A <see cref="IResourceBuilder{T}"/> for the <see cref="AzureTableStorageResource"/>.</returns>
```
Comment (JamesNK) [inline src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs:189]:
```suggestion
    /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureTableStorageResource"/>.</returns>
```

File: src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs [hunk 6]
```diff
@@ -196,11 +196,11 @@ public static IResourceBuilder<AzureTableStorageResource> AddTables(this IResour
...
     /// <summary>
-    /// TODO: Doc Comments
+    /// Creates a builder for the <see cref="AzureQueueStorageResource"/> which can be referenced to get the Azure Storage queues endpoint for the storage account.
     /// </summary>
-    /// <param name="builder"></param>
-    /// <param name="name"></param>
-    /// <returns></returns>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>/</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns>A <see cref="IResourceBuilder{T}"/> for the <see cref="AzureQueueStorageResource"/>.</returns>
```
Comment (JamesNK) [inline src/Aspire.Hosting.Azure/Extensions/AzureStorageExtensions.cs:203]:
```suggestion
    /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureQueueStorageResource"/>.</returns>
```

File: src/Aspire.Hosting/ApplicationModel/EndpointReference.cs [hunk 7]
```diff
@@ -153,9 +153,9 @@ public class EndpointReferenceExpression(EndpointReference endpointReference, En
     /// <summary>
     /// Gets the value of the property of the endpoint.
     /// </summary>
-    /// <param name="cancellationToken"></param>
-    /// <returns></returns>
-    /// <exception cref="InvalidOperationException"></exception>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/>.</param>
```
Comment (JamesNK) [inline src/Aspire.Hosting/ApplicationModel/EndpointReference.cs:156]: Some places say "a cancellation token", others say "the cancellation token". Should pick one and be consistent.

File: src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs [hunk 8]
```diff
@@ -90,8 +90,8 @@ public static bool TryGetEndpoints(this IResource resource, [NotNullWhen(true)]
     /// <summary>
     /// Gets the endpoints for the specified resource.
     /// </summary>
-    /// <param name="resource"></param>
-    /// <returns></returns>
+    /// <param name="resource">The <see cref="IResourceWithEndpoints"/> which contains <see cref="EndpointAnnotation"/> annotations.</param>
+    /// <returns>A enumeration of <see cref="EndpointReference"/> based on the <see cref="EndpointAnnotation"/> annotations from the resources' <see cref="IResource.Annotations"/> collection.</returns>
```
Comment (JamesNK) [inline src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs:94]:
```suggestion
    /// <returns>An enumeration of <see cref="EndpointReference"/> based on the <see cref="EndpointAnnotation"/> annotations from the resources' <see cref="IResource.Annotations"/> collection.</returns>
```

**Comment by davidfowl** [comment]
cc @normj

File: src/Aspire.Hosting.AWS/CloudFormation/CloudFormationStackExecutor.cs [hunk 9]
```diff
@@ -386,8 +386,8 @@ private async Task<bool> WaitForChangeSetBeingAvailableAsync(string changeSetId,
     /// <summary>
     /// Wait for the CloudFormation stack to get to a stable state after creating or updating the stack.
     /// </summary>
-    /// <param name="minTimeStampForEvents"></param>
-    /// <param name="cancellationToken"></param>
+    /// <param name="minTimeStampForEvents">Minimum timestamp for events.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
```
Comment (JamesNK) [inline src/Aspire.Hosting.AWS/CloudFormation/CloudFormationStackExecutor.cs:390]:
```suggestion
    /// <param name="cancellationToken">A <see cref="CancellationToken"/>.</param>
```

**Review by JamesNK — APPROVED** [review]

**Pull request merged**